community-release/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/google/go-github/v50/github"
+	"github.com/grafana/grafana-github-actions-go/pkg/changelog"
+	"github.com/grafana/grafana-github-actions-go/pkg/community"
+	"github.com/grafana/grafana-github-actions-go/pkg/toolkit"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+)
+
+const inputCommunityAPIKey = "COMMUNITY_API_KEY"
+const inputCommunityAPIUsername = "COMMUNITY_API_USERNAME"
+const inputCommunityCategoryID = "COMMUNITY_CATEGORY_ID"
+const inputCommunityBaseURL = "COMMUNITY_BASE_URL"
+const inputVersion = "VERSION"
+
+func main() {
+	logger := zerolog.New(zerolog.ConsoleWriter{Out: os.Stderr}).With().Timestamp().Logger()
+	ctx := context.Background()
+	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
+	defer cancel()
+	ctx = logger.WithContext(ctx)
+
+	var repo string
+	var doPreview bool
+	var listInputs bool
+
+	pflag.StringVar(&repo, "repo", os.Getenv("GITHUB_REPOSITORY"), "owner/repo pair for a repository on GitHub")
+	pflag.BoolVar(&doPreview, "preview", false, "Only determine the milestone but don't set it")
+	pflag.BoolVar(&listInputs, "list-inputs", false, "Show a list of all available inputs")
+	pflag.Parse()
+
+	logger.Info().Msgf("Operating inside %s", repo)
+
+	tk, err := toolkit.Init(
+		ctx,
+		toolkit.WithRegisteredInput(inputVersion, "Version number to generate the changelog for"),
+		toolkit.WithRegisteredInput(inputCommunityAPIKey, "API token for the Discourse community"),
+		toolkit.WithRegisteredInput(inputCommunityAPIUsername, "API username for the Discourse community"),
+		toolkit.WithRegisteredInput(inputCommunityCategoryID, "Discourse category ID for the changelog post"),
+		toolkit.WithRegisteredInput(inputCommunityBaseURL, "URL where the Discourse community can be found"),
+	)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("Failed to initialize toolkit")
+	}
+
+	if listInputs {
+		tk.ShowInputList()
+		return
+	}
+	defer func() {
+		if err := tk.SubmitUsageMetrics(ctx); err != nil {
```

Comment:
Not really sure if anyone still cares about that tbh 😅 . But it's a good thought!

Reply:
Yeah, no idea. It is in the original actions and so I thought I should re-implement it here 🙂